sdk/storage/azure-storage-common/inc/azure/storage/common/crypt.hpp

```diff
@@ -3,17 +3,21 @@
 
 #pragma once
 
+#include <azure/core/base64.hpp>
+
 #include <cstdint>
 #include <string>
 #include <vector>
 
 namespace Azure { namespace Storage {
 
+  namespace Internal {
+    inline std::string Base64EncodeText(const std::string& text)
+    {
+      return Azure::Core::Base64Encode(std::vector<uint8_t>(text.begin(), text.end()));
+    }
+  } // namespace Internal
   std::string Base64Encode(const std::vector<uint8_t>& data);
-  inline std::string Base64Encode(const std::string& text)
-  {
-    return Base64Encode(std::vector<uint8_t>(text.begin(), text.end()));
-  }
```

Comment:
What's the motivation of adding the suffix `Text` and moving it into `Internal` namespace? Why cannot this overloaded function be moved to `Azure::Core::` together with the other 2 functions?

Reply:
This is more of a convenience wrapper around the underling Base64 encode method and we shouldn't expose such things to the customer unless there is a strong reason to. I don't expect other services and client libraries to call or use such an API, and is storage specific.

In this particular case, Base64 is an encoding from binary data to string, and the encode method accepting a string parameter is confusing. Ideally, this internal method would be removed and all the call sites use the `vector<uint8_t>` based overload.

I added `Text` to reconcile this oddity and make it explicit that this isn't the "expected" encode method.

Reply:
ok